pkg/remote/kyma_synchronization_context.go

```diff
@@ -174,11 +225,24 @@ func (c *KymaSynchronizationContext) CreateOrFetchRemoteKyma(
 
 	if meta.IsNoMatchError(err) {
 		recorder.Event(kyma, "Normal", err.Error(), "CRDs are missing in SKR and will be installed")
+	}
 
-		if err := c.CreateOrUpdateCRD(ctx, v1beta2.KymaKind.Plural()); err != nil {
+	if meta.IsNoMatchError(err) || err == nil {
+		var kcpCrd, skrCrd *v1extensions.CustomResourceDefinition
```

Comment:
The follow logic you already used in both remote catalog sync and remote kyma sync, then you should consider extract them in to a shared function, and used in both place.

The logic I mean CreateOrUpdateCRD -> check annotation with the crd generation -> patch remote crd